code-challenges/Regular-Exp-Part1/challenges-04.test.js

```diff
@@ -0,0 +1,231 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1
+
+Write a function named isNum that takes in a string or number of any length. This function should use a regular expression pattern to return true if the input contains a number, and false if the input does not contain a number.
+
+For example:
+12345 returns true
+'12345' returns true
+'h3llo world' returns true
+'hello world' returns false
+------------------------------------------------------------------------------------------------ */
+
+const isNum = (input) => {
+  // Solution code here...
+  let isNumber = /[0-9]/g;
+  return isNumber.test(input);
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named isCapitalized that takes in a string. This function should use a regular expression pattern to match all words that begin with a capital letter. It should only match words, not punctuation.
+
+Return an array containing all the matches.
+------------------------------------------------------------------------------------------------ */
+
+const isCapitalized = (str) => {
+  // Solution code here...
+  
+  const capWord = /\b[A-Z]\w*/g;
+  const matches = str.match(capWord)
+  var array = [];
+  return matches ? matches :array;
+
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named citiesAtoJ that takes in an array of city names and uses a regular expression pattern to return a new array containing any cities that begin with the letters A through J, inclusive.
+------------------------------------------------------------------------------------------------ */
+
+const citiesAtoJ = (arr) => {
+//   // Solution code here...
+
+const cityAJ = /^[A-J]\w*/g;
+const cityArray = [];
+arr.forEach(element => {
+if (element.match(cityAJ)) 
+  cityArray.push(element);
+});
+return cityArray;
+};
+
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+You have created a game application and begin by asking users an easy question: In which month is Halloween?
+
+Write a function named matchMonth which uses a regular expression pattern to match any of these inputs: October, Oct, october, oct
+
+If the user enters any of these four inputs, return true. For any other input, return false.
+
+Do not use the vertical bar (pipe) in your pattern.
+------------------------------------------------------------------------------------------------ */
+
+const matchMonth = (input) => {
+  // Solution code here...
+
+  const monthm = /^[Oo]ct(ober)?$/;                 // by internet 
```

Comment:
"by internet" ... be more specific, please. I'd like to see the actual source where you found some help. You must always attribute your sources, as they can take action if they find their code somewhere.